src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,139 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        User firstUser = new User("username1", "password", 18);
+        User secondUser = new User("username2", "123456", 38);
+        Storage.people.add(firstUser);
+        Storage.people.add(secondUser);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_existingLogin_notOk() {
+        User sameLoginFirstUser = new User("username1", "password123", 18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(sameLoginFirstUser);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User loginIsFiveChar = new User("abcde", "password123", 34);
+        User loginIsTwoChar = new User("ab", "password123", 34);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(loginIsFiveChar);
+        });
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(loginIsTwoChar);
+        });
+    }
+
+    @Test
+    void register_zeroCharLogin_notOk() {
+        User user = new User("", "password123", 34);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User passwordFiveChar = new User("username", "12345", 34);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(passwordFiveChar);
+        });
+    }
+
+    @Test
+    void register_zeroLengthPassword_notOk() {
+        User user = new User("username", "", 34);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_youngAge_notOk() {
+        User ageTenYears = new User("username", "password", 10);
+        assertThrows(RegistrationException.class, () -> {
```

Comment:
corner case 17 is not tested

![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/74160013-2978-4278-a01a-28324dfe87d9)